src/opening-period/time-span/TimeSpan.test.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ReactNode } from 'react';
+import { act } from 'react-dom/test-utils';
+import { fireEvent, render, RenderResult } from '@testing-library/react';
+import { ArrayField, FormProvider, useForm } from 'react-hook-form';
+import { datePeriod } from '../../../test/fixtures/api-date-period';
+import { TimeSpanFormFormat } from '../../common/lib/types';
+import TimeSpan from './TimeSpan';
+
+function renderWithReactHookForm(
+  component: JSX.Element,
+  { defaultValues = {} } = {}
+): RenderResult {
+  const Wrapper = ({ children }: { children?: ReactNode }): JSX.Element => {
+    const methods = useForm({ defaultValues });
+    return <FormProvider {...methods}>{children}</FormProvider>;
+  };
+
+  return {
+    ...render(component, { wrapper: Wrapper }),
+  };
+}
+
+describe(`<TimeSpan />`, () => {
```

Comment:
Should all the timespan tests locate in the same place?

Reply:
This is a good question. I think they all can't be in this file, since some of the validations require sending the form. And sending is not available if only TimeSpan component is rendered. So maybe validation tests should be with the full form, and other features could be tested by individual components?

Reply:
There might still be some tests that could be here. I think we need some conventions for organizing tests.
I'm worried about test overlap; tests are also liability 🤔

Reply:
Maybe it could be something like this:
- rendering tests (including conditional rendering, ui changes) -> component tests
- save tests, validation tests -> form tests
- page rendering -> page tests